src/pages/item/item-actions/ItemActions.tsx

```diff
@@ -21,29 +31,98 @@ export default function ItemActions(props: { itemId: string | number }) {
 				"Can't load Downstream References - data still loading or failed to do so."
 			);
 		}
+		
+	};
+
+	//get all downstreamRefs and associations and then call in the miro.api a method which creates all connectors
+	// { downstreamRefs: [number], associations: [{associationId: number, targetItemId: number}] }
+	React.useEffect(() => {
+		async function fetchData() {
+			if (data && (data.downstreamReferences.length || data.outgoingAssociations.length)) {
+			var downstreamRefs: React.SetStateAction<number[]> = [];
+			var associations: ((prevState: never[]) => never[]) | { associationId: number; targetItemId: number; }[] = [];
+			if (data.downstreamReferences.length) {
+				data.downstreamReferences.forEach(function (downstreamReference) {
+				downstreamRefs.push(downstreamReference.itemRevision.id);
+				});
+			}
+			if (data.outgoingAssociations.length) {
+				data.outgoingAssociations.forEach(function (outgoingAssociation) {
+				var association = {associationId: outgoingAssociation.id, targetItemId: outgoingAssociation.itemRevision.id};
+				associations.push(association);
+				});
+			}
+
+			//check if downstreamReferences or outgoingAssociations exist on the board
+			const amountOfCardsOnBoard = await calculateAmountOfCardsOnBoard();
+
+			if(amountOfCardsOnBoard){
+				setShowDependenciesDisabled(false)
+			}
+			else{
+				setShowDependenciesDisabled(true)
+			}
+
+			setDownstreamRefs(downstreamRefs);
+			setAssociations(associations);
+			}
+		}
+	  
+		fetchData();
+	}, [data]);
+	  
+
+	const showDependenciesHandler = async () => {
+		if (data && !showDependenciesDisabled) {
+			const startCardIds = await getAppCardId(parseInt(props.itemId));
+			const startCardId = startCardIds[0]
+			await createConnectorsForDownstreamRefsAndAssociation(startCardId, downstreamRefs, associations);
+
+		  } else {
+			console.warn("Can't load Associations - data still loading or failed to do so.");
+		  }
 	};
 
+	const calculateAmountOfCardsOnBoard = async () => {
```

Comment:
Inadequate name (function & the `var` in it), I'd argue. The number isn't what it claims to be, is it?

Reply:
changed it to calculateAmountOfRelationsOnBoard

Reply:
You're still using a `var` there.  
That might be part of the issue with the bug I showed you, where the displayed number of relations on board for some item was way too high.